docs/02.API-REFERENCE.md

```diff
@@ -6558,6 +6586,71 @@ jerry_create_arraybuffer (jerry_length_t size);
 - [jerry_value_is_arraybuffer](#jerry_value_is_arraybuffer)
 - [jerry_release_value](#jerry_release_value)
 
+# jerry_arraybuffer_set_external_free_callback
+
+**Summary**
+
+Sets the global callback which is called when an external arraybuffer is released.
+It is recommended to set this function before the first external arraybuffer is created.
+
+*Additional info*:
+- When the callback is NULL, no function is called when an external arraybuffer is freed.
+
+**Prototype**
+
+```c
+void jerry_arraybuffer_set_external_free_callback (jerry_external_arraybuffer_free_callback_t callback_p);
+```
+
+- `callback_p` - callback which is called when an external arraybuffer is freed.
+
+*New in version [[NEXT_RELEASE]]*.
+
+**Example**
+
+[doctest]: # ()
+
+```c
+#include "jerryscript.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static void
+external_arraybuffer_free_callback (uint32_t length, /**< length of buffer */
+                                    void *buffer, /**< buffer to free (if needed) */
+                                    void *user_p) /**< user pointer */
+{
+  free (buffer);
+  printf ("External arraybuffer is freed!\n");
+}
+
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+  
+  jerry_arraybuffer_set_external_free_callback (external_arraybuffer_free_callback);
+  
+  const uint32_t buffer_size = 15;
+
+  uint8_t *buffer_p = (uint8_t *) malloc (sizeof (uint8_t) * buffer_size);
+  memset (buffer_p, 51 /* default value */, buffer_size);
+  
+  jerry_value_t external_arraybuffer = jerry_create_arraybuffer_external (buffer_size, buffer_p, NULL);
+  
+  /* The external_arraybuffer_free_callback is called. */
```

Comment:
comment: The external_arraybuffer_free_callback is called eventually by the garbage collector.